projects/library/button/src/button.component.scss

```diff
@@ -92,39 +92,53 @@ ts-button {
     transition: var(--button-color-transition), var(--button-background-transition), var(--button-shadow-transition);
   }
 
+  // TODO: Sass area of possible themes should be available to all components from a global location
+  // Define all possible themes
+  $themes: primary accent warn;
+  // Loop through using each theme name to create styles
+  @each $theme in $themes {
+    // Theme class sets the colors for standard buttons
```

Comment:
Similar pattern I saw in chip component as well... is it something that can be refactored later?

Reply:
Yeah, you'll likely see it in most components that use themes. It's possible we could abstract it out, but I'm not really sure how easy it would be since the theme implementation is very component specific. (ie what elements within a component get styled by themes)

Open to any ideas though 👍